### PR TITLE
Refactor immunization extraction

### LINKED ISSUE #488: Update the save vaccine method and also post extra fields for Covid-19 Vaccination in Covax demo app
Replace manual mapping with resource mapper and simplify

```
  private fun handleImmunizationResult(response: QuestionnaireResponse) {
    // TODO Replace manual mapping with resource mapper and simplify

    val questionnaire =
      questionnaireViewModel.loadQuestionnaire(detailView.vaccineQuestionnaireIdentifier)

    covaxListViewModel.getPatientItem(clientIdentifier).observe(this) {
      val immunization =
        ResourceMapper.extract(questionnaire, response).entry[0].resource as Immunization
      immunization.id = UUID.randomUUID().toString()
      immunization.recorded = Date()
      immunization.status = Immunization.ImmunizationStatus.COMPLETED
      immunization.vaccineCode =
        CodeableConcept().apply {
          this.text = response.item[0].answer[0].valueCoding.code
          this.coding = listOf(response.item[0].answer[0].valueCoding)
        }
      immunization.occurrence = DateTimeType.today()
      immunization.patient = Reference().apply { this.reference = "Patient/$clientIdentifier" }

      immunization.protocolApplied =
        listOf(
          Immunization.ImmunizationProtocolAppliedComponent().apply {
            val currentDoseNumber = it.vaccineSummary?.doseNumber ?: 0
            this.doseNumber = PositiveIntType(currentDoseNumber + 1)
          }
        )
      showVaccineRecordDialog(immunization, it)
    }
  }
```

Update the Immunization resource being posted to the server 

- [ ] 1. vaccineCode - keep static based on the sample shared, to be updated from a plan later with full details , change the text files based on the vaccine selected   "text": "Moderna"
- [x] 2. manufacturer - keep static based on the sample shared, to be updated from a plan later with full details 
- [x] 3. lotNumber - add a random number, see sample resource below 
- [x] 4. expirationDate - add a random date in the future, see sample resource below 
- [x] 5. encounter Reference  - add a reference to an encounter,   create a new encounter for an Immunization event. This will require multiple resource submissions.
- [x] 6. occurrenceDateTime - same as recorded DateTimeType.today() 
- [ ] 7. location - add a reference to a location,  see sample resource below. Pick the location of the vaccination centre or from the Practioner details synced at Login
- [x] 8. doseQuantity - update, see sample resource below 
- [ ] 9. performer - add a reference to a practitioner, see sample resource below. Query from the Practioner details synced at Login
- [x] 10. protocolApplied - update to show  the series, authority, targetDisease, doseNumberPositiveInt seriesDosesPositiveInt,  see sample resource below 

Sample Immunization resource for Dose 1

```
{
  "resourceType": "Immunization",
  "status": "completed",
  "vaccineCode": {
    "coding": [
      {
        "system": "http://snomed.info/sct",
        "code": "1119349007",
        "display": "SARSCoV2  mRNA vaccine"
      }
    ],
    "text": "Moderna"
  },
  "manufacturer": {
    "identifier": {
      "system": "http://worldhealthorganization.github.io/ddcc/CodeSystem/DDCC-Example-Test-CodeSystem"
    },
    "display": "BioTech"
  },
  "lotNumber": "PT123F",
  "expirationDate": "2022-12-15",
  "patient": {
    "reference": "Patient/969"
  },
  "encounter": {
    "reference": "Encounter/974"
  },
  "occurrenceDateTime": "2021-01-01",
  "recorded": "2021-01-01T00:16:52+03:00",
  "location": {
    "display": "Vaccination Site",
    "reference": "Location/971"
  },
  "doseQuantity": {
    "value": 2,
    "system": "http://unitsofmeasure.org",
    "code": "mg"
  },
  "performer": [ {
    "actor": {
      "type": "Practitioner",
      "identifier": {
        "value": "http://www.acme.org/practitioners/23"
      },
      "reference": "Practitioner/970"
    }
  } ],
  "protocolApplied": [
    {
      "series": "2-dose",
      "authority": {
        "type": "Organization",
        "identifier": {
          "value": "wA69g8VD512TfTTdkTNSsG"
        }
      },
      "targetDisease": [
        {
          "coding": [
            {
              "system": "http://snomed.info/sct",
              "code": "840539006",
              "display": "COVID 19"
            }
          ]
        }
      ],
      "doseNumberPositiveInt": 1,
      "seriesDosesPositiveInt": 2
    }
  ]
}
```

Sample Immunization resource for Dose 2

```
{
  "resourceType": "Immunization",
  "status": "completed",
  "vaccineCode": {
    "coding": [
      {
        "system": "http://snomed.info/sct",
        "code": "1119349007",
        "display": "SARSCoV2  mRNA vaccine"
      }
    ],
    "text": "Moderna"
  },
  "manufacturer": {
    "identifier": {
      "system": "http://worldhealthorganization.github.io/ddcc/CodeSystem/DDCC-Example-Test-CodeSystem"
    },
    "display": "BioTech"
  },
  "lotNumber": "PT123F",
  "expirationDate": "2022-12-15",
  "patient": {
    "reference": "Patient/969"
  },
  "encounter": {
    "reference": "Encounter/974"
  },
  "occurrenceDateTime": "2021-01-01",
  "recorded": "2021-01-01T00:16:52+03:00",
  "location": {
    "display": "Vaccination Site",
    "reference": "Location/971"
  },
  "doseQuantity": {
    "value": 2,
    "system": "http://unitsofmeasure.org",
    "code": "mg"
  },
  "performer": [ {
    "actor": {
      "type": "Practitioner",
      "identifier": {
        "value": "http://www.acme.org/practitioners/23"
      },
      "reference": "Practitioner/970"
    }
  } ],
  "protocolApplied": [
    {
      "series": "2-dose",
      "authority": {
        "type": "Organization",
        "identifier": {
          "value": "wA69g8VD512TfTTdkTNSsG"
        }
      },
      "targetDisease": [
        {
          "coding": [
            {
              "system": "http://snomed.info/sct",
              "code": "840539006",
              "display": "COVID 19"
            }
          ]
        }
      ],
      "doseNumberPositiveInt": 2,
      "seriesDosesPositiveInt": 2
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by ekigamba**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #488

**Type**
Choose one: Feature & Code health

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/eir/src/main/assets/form_configurations.json [hunk 1]
```diff
@@ -7,6 +7,6 @@
   {
     "form": "record-vaccine",
     "title": "Record Vaccine",
-    "identifier": "800"
+    "identifier": "2358"
```
Comment (ndegwamartin) [inline android/eir/src/main/assets/form_configurations.json:10]: This value is currently dynamically injected no?

Comment (ekigamba) [reply android/eir/src/main/assets/form_configurations.json:10]: Yes, but the origin of the value is this file

Comment (ndegwamartin) [reply android/eir/src/main/assets/form_configurations.json:10]: I see, so it can work with blank here?

**Review by ndegwamartin — DISMISSED** [review]

**Review by pld — APPROVED** [review]

File: android/eir/src/main/assets/form_configurations.json (hunk 1 above)
Comment (ekigamba) [reply android/eir/src/main/assets/form_configurations.json:10]: No, this is the source of the record-vaccine questionnaire identifier. If this is blank, then the app will always return null for the form

**Pull request merged**